autogpt_platform/backend/test/sdk/test_sdk_block_creation.py

```diff
@@ -0,0 +1,438 @@
+"""
+Tests for creating blocks using the SDK.
+
+This test suite verifies that blocks can be created using only SDK imports
+and that they work correctly without decorators.
+"""
+
+from typing import Optional
+
+import pytest
+
+from backend.sdk import (
+    APIKeyCredentials,
+    Block,
+    BlockCategory,
+    BlockOutput,
+    BlockSchema,
+    CredentialsMetaInput,
+    SchemaField,
+    SecretStr,
+)
+
+from ._config import test_api, test_service
+
+
+class TestBasicBlockCreation:
+    """Test creating basic blocks using the SDK."""
+
+    @pytest.mark.asyncio
+    async def test_simple_block(self):
+        """Test creating a simple block without any decorators."""
+
+        class SimpleBlock(Block):
+            """A simple test block."""
+
+            class Input(BlockSchema):
+                text: str = SchemaField(description="Input text")
+                count: int = SchemaField(description="Repeat count", default=1)
+
+            class Output(BlockSchema):
+                result: str = SchemaField(description="Output result")
+
+            def __init__(self):
+                super().__init__(
+                    id="simple-test-block",
+                    description="A simple test block",
+                    categories={BlockCategory.TEXT},
+                    input_schema=SimpleBlock.Input,
+                    output_schema=SimpleBlock.Output,
+                )
+
+            async def run(self, input_data: Input, **kwargs) -> BlockOutput:
+                result = input_data.text * input_data.count
+                yield "result", result
+
+        # Create and test the block
+        block = SimpleBlock()
+        assert block.id == "simple-test-block"
+        assert BlockCategory.TEXT in block.categories
+
+        # Test execution
+        outputs = []
+        async for name, value in block.run(
+            SimpleBlock.Input(text="Hello ", count=3),
+        ):
+            outputs.append((name, value))
+        assert len(outputs) == 1
+        assert outputs[0] == ("result", "Hello Hello Hello ")
+
+    @pytest.mark.asyncio
+    async def test_block_with_credentials(self):
+        """Test creating a block that requires credentials."""
+
+        class APIBlock(Block):
+            """A block that requires API credentials."""
+
+            class Input(BlockSchema):
+                credentials: CredentialsMetaInput = test_api.credentials_field(
+                    description="API credentials for test service",
+                )
+                query: str = SchemaField(description="API query")
+
+            class Output(BlockSchema):
+                response: str = SchemaField(description="API response")
+                authenticated: bool = SchemaField(description="Was authenticated")
+
+            def __init__(self):
+                super().__init__(
+                    id="api-test-block",
```

Comment:
this should also not be able to be registered

Reply:
out of scope of the sdk changes as this is going into functionality already part of the system